airsonic-main/src/main/java/org/airsonic/player/service/metadata/JaudiotaggerParser.java

```diff
@@ -96,8 +99,69 @@ public MetaData getRawMetaData(File file) {
+                            && ID3v24Tag.REVISION == id3Tag.getRevision()) {
+
+                        audioFile.getTag().getFields().forEachRemaining(f -> {
+                            switch (f.getId()) {
+                                case ID3v24Frames.FRAME_ID_ALBUM:
+                                    if (null == metaData.getAlbumName()) {
+                                        metaData.setAlbumName(f.toString());
+                                    }
+                                    break;
+                                case ID3v24Frames.FRAME_ID_TITLE:
+                                    if (null == metaData.getTitle()) {
+                                        metaData.setTitle(f.toString());
+                                    }
+                                    break;
+                                case ID3v24Frames.FRAME_ID_YEAR:
+                                    if (null == metaData.getYear()) {
+                                        metaData.setYear(parseYear(f.toString()));
+                                    }
+                                    break;
+                                case ID3v24Frames.FRAME_ID_GENRE:
+                                    if (null == metaData.getGenre()) {
+                                        metaData.setGenre(mapGenre(f.toString()));
+                                    }
+                                    break;
+                                case ID3v24Frames.FRAME_ID_SET:
+                                    if (null == metaData.getDiscNumber()) {
+                                        metaData.setDiscNumber(parseInteger(f.toString()));
+                                    }
+                                    break;
+                                case ID3v24Frames.FRAME_ID_TRACK:
+                                    if (null == metaData.getTrackNumber()) {
+                                        metaData.setTrackNumber(parseTrackNumber(f.toString()));
+                                    }
+                                    break;
+                                case ID3v24Frames.FRAME_ID_ARTIST:
+                                    if (null == metaData.getArtist() && null == songArtist) {
+                                        metaData.setArtist(f.toString());
+                                    }
+                                    break;
+                                case ID3v24Frames.FRAME_ID_ACCOMPANIMENT:
+                                    if (null == metaData.getAlbumArtist()) {
+                                        metaData.setAlbumArtist(f.toString());
+                                    }
+                                    break;
+                                default:
+                                    break;
+                            }
+                        });
+                    }
+                }
+                if(StringUtils.isBlank(metaData.getArtist())) {
```

Comment:
Shouldn't this be in an `else`?

Reply:
> Protip: you can [rewrite your commit history](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History), and then force-push, instead of opening several pull-requests :)

 Sorry... You're absolutely right.

> Shouldn't this be in an `else`?

Generally it is very rare, but there may be cases where Artist does not exist in either id3v2.1 or id3v2.4 field.
However certainly, it may be that there is not enough consideration about the point indicated.

When the relationship between Artist and AlbumArtist is as follows.
(First of all, it is necessary to confirm this specification.)

![image](https://user-images.githubusercontent.com/27724847/55908530-a80c7200-5c14-11e9-8cf4-1f221428bfa6.png)
Case 3 may now have a bad implementation.

Reply:
Could you add links to the specifications in the code that back/approves your changes ?

Reply:
Certainly.